Core.lua

```diff
@@ -453,11 +454,13 @@ f:SetScript("OnEvent", function(self, event, arg1, arg2)
 	Core:Debug(event);
 
 	if event == "PLAYER_REGEN_DISABLED" then
-		if containerFrame and containerFrame:IsShown() then
+		if GOW.DB.profile.hideInCombat == true and containerFrame and containerFrame:IsShown() then
 			containerFrame:Hide();
```

Comment:
no need == true
you can move this whole if at the end of event checks with elseif
return can be removed.